setuptools/build_meta.py

```diff
@@ -209,9 +210,13 @@ def build_wheel(self, wheel_directory, config_settings=None,
                                          wheel_directory, config_settings)
 
     def build_sdist(self, sdist_directory, config_settings=None):
-        return self._build_with_temp_dir(['sdist', '--formats', 'gztar'],
-                                         '.tar.gz', sdist_directory,
-                                         config_settings)
+        orig, sdist.include_pyproject_toml = sdist.include_pyproject_toml, True
```

Comment:
I'd love for this not to be monkey-patched, but assuming that's how it must be, this patch should be a one-line call to encapsulate the monkeypatch behavior and decouple it from the underlying call. Something like `with self._enable_pyproject_toml():`.